packages/quantic/force-app/main/default/lwc/quanticHeadlessLoader/quanticHeadlessLoader.js

```diff
@@ -158,20 +174,33 @@ function setComponentInitialized(element, engineId) {
     throw new Error('Fatal Error: Component was not registered before initialization');
   }
   component.initialized = true;
-  if (areAllComponentsInitialized(engineId)) {
-    debounceInitialSearch(engineId);
+  if (window.coveoHeadless[engineId].initializedCallback && areAllComponentsInitialized(engineId)) {
+    debounceInitializedCallback(engineId);
   }
 }
 
 /**
  * Returns headless engine promise.
  * @param {string} engineId The id of the engine.
  */
-async function getHeadlessEngine(engineId) {
-  if (!window.coveoHeadless[engineId].engine) {
-    window.coveoHeadless[engineId].engine = initEngine(engineId);
+function getHeadlessEnginePromise(engineId) {
+  if (!window.coveoHeadless[engineId].enginePromise) {
+    window.coveoHeadless[engineId].enginePromise = initEngine(engineId).then((engine) => {
+      window.coveoHeadless[engineId].bindings.engine = engine
+      return engine;
+    });
+  }
+  return window.coveoHeadless[engineId].enginePromise;
+}
+
+/**
+ * Returns bindings object for specified engineId.
+ * @param {string} engineId The id of the engine.
+ */
+function getHeadlessBindings(engineId) {
+  return {
+    engine: window.coveoHeadless[engineId].bindings,
```

Comment:
Is this just the wrong name?

`getHeadlessBindings` function that returns an object containing the bindings, but under `engine` ?

Reply:
ah yeah refactoring error